artifactory/utils/projectconfig.go

```diff
@@ -0,0 +1,118 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/jfrog/jfrog-cli-go/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/spf13/viper"
+	"path/filepath"
+	"reflect"
+)
+
+const (
+	PROJECT_CONFIG_RESOLVER_PREFIX = "resolver"
+	PROJECT_CONFIG_DEPLOYER_PREFIX = "deployer"
+	PROJECT_CONFIG_REPO            = "repo"
+	PROJECT_CONFIG_SERVER_ID       = "serverId"
+)
+
+type ProjectType int
+
+const (
+	GO ProjectType = iota
+	PIP
+)
+
+var ProjectTypes = []string{
+	"go",
+	"pip",
+}
+
+func (projectType ProjectType) String() string {
+	return ProjectTypes[projectType]
+}
+
+type Repository struct {
+	Repo     string `yaml:"repo,omitempty"`
+	ServerId string `yaml:"serverId,omitempty"`
+}
+
+type RepositoryConfig struct {
+	targetRepo string
+	rtDetails  *config.ArtifactoryDetails
+}
+
+// If configuration file exists in the working dir return its path, otherwise return the global configuration file path
```

Comment:
// ... in the working dir ... -->// ... in the working dir or in one of its parent dirs...